test/index.js

```diff
@@ -766,6 +766,136 @@ describe('def', function() {
     eq(div(4, 2), 2);
   });
 
+  it('supports the "Integer" type', function() {
+    var def = $.create($.env.concat([$.Integer]));
+
+    //  inc :: Integer -> Integer
+    var inc = def('inc',
+                  {},
+                  [$.Integer, $.Integer],
+                  function(x) { return x + 1; });
+
+    throws(function() { inc(3.14); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type Integer ' +
+                   'as its first argument; received 3.14'));
+
+    throws(function() { inc(9007199254740992); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type Integer ' +
+                   'as its first argument; received 9007199254740992'));
+
+    throws(function() { inc(new Number(-9007199254740992)); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type Integer ' +
+                   'as its first argument; received ' +
+                   'new Number(-9007199254740992)'));
+
+    throws(function() { inc(new Number(-9007199254740992)); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type Integer ' +
+                   'as its first argument; received ' +
+                   'new Number(-9007199254740992)'));
+
+    eq(inc(1), 2);
+  });
+
+  it('supports the "NonZeroInteger" type', function() {
+    var def = $.create($.env.concat([$.ValidNumber, $.NonZeroInteger]));
+
+    //  div :: ValidNumber -> NonZeroInteger -> ValidNumber
+    var div = def('div',
+                  {},
+                  [$.ValidNumber, $.NonZeroInteger, $.ValidNumber],
+                  function(a, b) { return a / b; });
+
+    throws(function() { div(3, 0); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroInteger ' +
+                   'as its second argument; received 0'));
+
+    throws(function() { div(3, new Number(0)); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroInteger ' +
+                   'as its second argument; received new Number(0)'));
+
+    throws(function() { div(3, -0); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroInteger ' +
+                   'as its second argument; received -0'));
+
+    throws(function() { div(3, new Number(-0)); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroInteger ' +
+                   'as its second argument; received new Number(-0)'));
+
+    eq(div(4, 2), 2);
+  });
+
+  it('supports the "PositiveInteger" type', function() {
+    var def = $.create($.env.concat([$.PositiveInteger]));
+
+    //  inc :: PositiveInteger -> PositiveInteger
+    var inc = def('inc',
+                  {},
+                  [$.PositiveInteger, $.PositiveInteger],
+                  function(x) { return x + 1; });
+
+    throws(function() { inc(-1.5); },
```

Comment:
`-1.5` is neither positive nor an integer. Since all the invalid input values are currently negative, we're not necessarily testing the "integer" check. We could use `1.5` here instead.